mewarpx/mewarpx/emission.py

```diff
@@ -1186,6 +1182,108 @@ def get_normals(self, x, y, z):
         return normals
 
 
+class ZDiscEmitter(Emitter):
+
+    """This injects over an x-y disc rather than a rectangle."""
+    geoms = ['RZ']
+
+    def __init__(self, conductor, T, inner_emission_radius=None,
+                 outer_emission_radius=None, transverse_fac=1.0, **kwargs):
+        """Initialize an emitter for a disc (circular) cathode.
+
+        Arguments:
+            conductor (warp.Assembly): Conductor object, used to obtain work
+                function. Can later grab other variables from this conductor.
+            T (float): Temperature in Kelvin for the emitter; determines
+                velocities.
+            inner_emission_radius (float): Inner radius of the disc (in meters)
+                for particles to be emitted from. Default mwxrun.rmin.
+            outer_emission_radius (float): Outer radius of the disc (in meters)
+                for particles to be emitted from. Default mwxrun.rmax.
+            transverse_fac (float): Scale the transverse energy distribution by
+                this factor. Default 1. See
+                :func:`metools.runtools.get_velocities` for details.
+            kwargs (dict): Any other keyword arguments supported by the parent
+                Emitter constructor (such as "use_Schottky" or
+                "emission_type").
+
+        Notes:
+            The center of the disc is always x = y = 0 at present.
+        """
+        # Default initialization
+        super(ZDiscEmitter, self).__init__(T=T, conductor=conductor, **kwargs)
+
+        self.z = conductor.z
+        self.zsign = conductor.zsign
+        self.transverse_fac = transverse_fac
+
+        # Save input parameters
+        if inner_emission_radius is None:
+            inner_emission_radius = mwxrun.rmin
+        self.inner_emission_radius = inner_emission_radius
+        if outer_emission_radius is None:
+            outer_emission_radius = mwxrun.rmax
+        self.outer_emission_radius = outer_emission_radius
+
+        if mwxrun.rmin > 0:
+            raise RuntimeError("r_min > 0 not yet implemented for ZDiscEmitter")
+
```

Comment:
Why is this? I don't see a problem in this class specifically.

Reply:
That is outdated. Thanks for catching!